### ISSUE TITLE
few storybook addons had dependency on prettier which break new syntax

### ISSUE THREAD

**Issue by rajzik** [storybookjs/storybook#13322]
**Describe the bug**

- `@storybook/addon-docs`
- `@storybook/addon-storysource`
- `@storybook/source-loader` 

depends on prettier currently in version `~2.0.5`, this can cause issue when using different version in project.
refer to [issue eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier/issues/359) for more info.

**To Reproduce**

1. Install storybook and essential addons
2. Install prettier with another version
3. Setup eslint with prettier and typescript
4. Try to use new types syntax from typescript

```tsx
export default function capitalize<T extends string>(text: T): `${Capitalize<T>}` {
  return (text.charAt(0).toUpperCase() + text.slice(1)) as `${Capitalize<T>}`;
}
```

**Expected behavior**
It would be great to have prettier as peer dependency and let developer decide which version to use.

**Screenshots**
![image](https://user-images.githubusercontent.com/10364836/100485583-eb289b00-3100-11eb-8bd4-ce76a7f3302b.png)

**System**

```
Environment Info:

  System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
  Binaries:
    Node: 15.2.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 7.0.12 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 87.0.4280.66
    Edge: Spartan (44.19041.423.0), Chromium (87.0.664.47)
  npmPackages:
    @storybook/addon-a11y: ^6.1.6 => 6.1.6
    @storybook/addon-actions: ^6.1.6 => 6.1.6
    @storybook/addon-docs: ^6.1.6 => 6.1.6
    @storybook/addon-essentials: ^6.1.6 => 6.1.6
    @storybook/addon-links: ^6.1.6 => 6.1.6
    @storybook/addon-storysource: ^6.1.6 => 6.1.6
    @storybook/addon-viewport: ^6.1.6 => 6.1.6
    @storybook/addons: ^6.1.6 => 6.1.6
    @storybook/components: ^6.1.6 => 6.1.6
    @storybook/react: ^6.1.6 => 6.1.6
    @storybook/theming: ^6.1.6 => 6.1.6
```

**Additional context**

**Comment by shilman** [comment]
Can we just upgrade prettier instead?

**Comment by BPScott** [comment]
This is a problem because @rajzik never depends upon `prettier` in their package.json. Thus it is a crapshoot about which one gets hoisted up to the top level and used by tools like eslint-plugin-prettier. As of right now he's ended up with prettier v2.0.5 from storybook's addons, rather than v2.2.0 from other packages.


For storybook updating to prettier 2.2 to support the new TS syntax iis a good bandaid.

Though this cropped up because the packages lock to `~2.0.5` so consumers currently easily dedupe to get down to a single prettier version. Perhaps those ranges should use `^` instead of `~`?

**Comment by rajzik** [comment]
@shilman it would be great to update this package and change from `~` to `^`. This would fix the issue for some time until version 3 of prettier.

**Comment by rajzik** [comment]
@BPScott direct dependency also resolves the issue, but we are using management layer which provides all the functionality that we need in few packages. So, we don't have to install packages directly.

**Comment by santialbo** [comment]
I'm having issues with this contantly which force me to reinstall packages and restart vscode. Is there a chance we could upgrade prettier before the 6.2.0 release?

**Comment by shilman** [comment]
@santialbo is it a breaking change for existing users?

**Comment by santialbo** [comment]
@shilman I don't think it would be a breaking change unless they were indirectly relying on the version installed by storybook.
However, to be sure I would need to know more about how prettier is being used in Storybook. If it just for prettifying code in some sort of "source" window, the rendered code would change as prettier introduced some changes in how the output is generated.
If prettier is also used to prettify TypeScript code, it would be interesting to upgrade as some of the newer typescript features require a more recent version of prettier.

**Comment by ndelangen** [comment]
This should be solved by storybook 7.0 beta
